name: issue bot
on:
    # schedule:
    #   - cron: "30 1 * * *"
    issues:
        types:
            - labeled

jobs:
    # close-issues:
    #   runs-on: ubuntu-latest
    #   permissions:
    #     issues: write
    #     pull-requests: write
    #   steps:
    #     - uses: actions/stale@v5
    #       with:
    #         days-before-issue-stale: 30
    #         days-before-issue-close: 7
    #         stale-issue-label: "stale"
    #         stale-issue-message: "This issue is stale because it has been open for 30 days with no activity."
    #         close-issue-message: "This issue was closed because it has been inactive for 7 days since being marked as stale."
    #         exempt-issue-labels: "help wanted"
    #         days-before-pr-stale: -1
    #         days-before-pr-close: -1
    #         repo-token: ${{ secrets.GITHUB_TOKEN }}
    issue-invalid:
        name: close invalid issue
        if: github.event.label.name == 'invalid'
        runs-on: ubuntu-latest
        permissions:
            issues: write
            pull-requests: write
        steps:
            - uses: actions-cool/issues-helper@v3
              with:
                  actions: close-issue, create-comment
                  token: ${{ secrets.GITHUB_TOKEN }}
                  body: |
                      Hello @${{ github.event.issue.user.login }}. This issue is marked as `invalid` and closed. Please make sure you are reporting an issue and following the issue template.
    issue-duplicate:
        name: close duplicate issue
        if: github.event.label.name == 'duplicate'
        runs-on: ubuntu-latest
        permissions:
            issues: write
            pull-requests: write
        steps:
            - uses: actions-cool/issues-helper@v3
              with:
                  actions: close-issue, create-comment
                  token: ${{ secrets.GITHUB_TOKEN }}
                  body: |
                      Hello @${{ github.event.issue.user.login }}. This issue is marked as `duplicate` and closed. Please make sure you have searched to see if an issue already exists for the bug you encountered.
